Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/FilterScreen.kt

```diff
@@ -0,0 +1,261 @@
+                            )
+                        }
+                    }
+                }
+                Row {
+                    FilterTitle(text = stringResource(id = R.string.max_calories))
+                    Text(
+                        text = stringResource(id = R.string.per_serving),
+                        style = MaterialTheme.typography.body2,
+                        color = JetsnackTheme.colors.brand,
+                        modifier = Modifier.padding(top = 5.dp, start = 10.dp)
+                    )
+                }
+                MaxCaloriesSlider()
+                FilterTitle(text = stringResource(id = R.string.lifestyle))
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(top = 12.dp, bottom = 28.dp),
+                    horizontalArrangement = Arrangement.Center
+                ) {
+                    FlowRow(
+                        modifier = Modifier.padding(horizontal = 4.dp)
+                    ) {
+                        lifeStyleFilters.forEach {
+                            FilterChip(
+                                filter = it,
+                                modifier = Modifier.padding(end = 5.dp, bottom = 8.dp)
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun SortFilters(onChanged: (Boolean) -> Unit) {
+    val sortFilters = SnackRepo.getSortFilters()
+    val defaultValue = SnackRepo.getSortDefault()
+    var sortState by remember { mutableStateOf(SnackRepo.getSortDefault()) }
+    sortFilters.forEach {
+        SortOption(
+            text = it.name,
+            icon = it.icon,
+            selected = sortState == it.name,
+            onClickOption = {
+                sortState = it.name
+                onChanged(it.name != defaultValue)
+            }
+        )
+    }
+}
+
+@Composable
+fun MaxCaloriesSlider() {
+    var sliderPosition by remember { mutableStateOf(0f) }
+
+    Slider(
+        value = sliderPosition,
+        onValueChange = {
+            sliderPosition = it
+        },
+        valueRange = 0f..300f,
+        steps = 5,
+        modifier = Modifier
+            .fillMaxWidth(),
+        colors = SliderDefaults.colors(
+            thumbColor = JetsnackTheme.colors.brand,
+            activeTrackColor = JetsnackTheme.colors.brand
+        )
+    )
+}
+
+@Composable
+fun FilterTitle(text: String,) {
+    Text(
+        text = text,
+        style = MaterialTheme.typography.h6,
+        color = JetsnackTheme.colors.brand,
+        modifier = Modifier.padding(bottom = 8.dp)
+    )
+}
+@Composable
+fun SortOption(
+    text: String,
+    icon: ImageVector?,
+    onClickOption: () -> Unit,
+    selected: Boolean
+) {
+    Row(
+        modifier = Modifier
+            .padding(top = 14.dp)
+            .clickable { onClickOption() }
```

Comment:
Would `Modifier.selectable` be a better fit here for a mutually exclusive group?

Reply:
Done